### PR TITLE
Improve HTTP download

### PR DESCRIPTION
**Pull request by ybiquitous**
We are currently using [`Net::HTTP`](https://ruby-doc.org/stdlib-2.6.5/libdoc/net/http/rdoc/Net/HTTP.html) to download a user's source code.

Because the `Net::HTTP` has own retry mechanism, this change aims to make our retrying more simple.

### CODE REVIEW

File: lib/runners/workspace/http.rb [hunk 1]
```diff
@@ -19,40 +19,58 @@ def prepare_head_source(dest)
...
+        use_ssl: true,
+        max_retries: max_retries,
+      }
+
+      Net::HTTP.start(uri.hostname, uri.port, http_options) do |http|
+        http.request_get(uri) do |response|
+          case response
+          when Net::HTTPSuccess
+            response.read_body(dest)
+          when Net::HTTPRedirection
```
Comment (ybiquitous) [inline lib/runners/workspace/http.rb]: [note] See also the link below about redirection:

https://ruby-doc.org/stdlib-2.6.5/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-Following+Redirection

**Comment by ybiquitous** [comment]
@yykamei This PR is in progress yet, but can you tell me your idea? Can I continue?

File: lib/runners/workspace/http.rb [hunk 2]
```diff
@@ -19,40 +19,62 @@ def prepare_head_source(dest)
     # @param dest [Pathname]
     # @param key [String, nil]
     def provision(uri, dest, key)
-      # When OpenSSL::Cipher::CipherError, probably the downloading is failed.
```
Comment (ybiquitous) [inline lib/runners/workspace/http.rb]: [question] Can `#decrypt` method raise `OpenSSL::Cipher::CipherError`?

https://github.com/sider/runners/blob/c8b77180b66344e868da3868efa20f05daca522c/test/workspace/http_test.rb#L19

Comment (yykamei) [reply lib/runners/workspace/http.rb]: I want to expect `Net::HTTP` to handle such an error with `max_retries`. Let's forget `CipherError`

**Comment by yykamei** [comment]
I think this is a nice refactoring.

**Pull request merged**